TestPromtail is flaky

On MacOS, about ~50% of the time I get:

```
time="2019-05-08T09:49:28+01:00" level=info msg="server listening on addresses" grpc="[::]:9095" http="[::]:80"
level=info msg="Adding target" key="{job=\"varlogs\", localhost=\"\"}"
level=info msg="start tailing file" path=/tmp/promtail_test_rTPtxgzxXg/logs/startupMarker.log
2019/05/08 09:49:33 Seeked /tmp/promtail_test_rTPtxgzxXg/logs/startupMarker.log - &{Offset:0 Whence:0}
level=info msg="start tailing file" path=/tmp/promtail_test_rTPtxgzxXg/logs/testSingle.log
2019/05/08 09:49:33 Seeked /tmp/promtail_test_rTPtxgzxXg/logs/testSingle.log - &{Offset:0 Whence:0}
level=info msg="start tailing file" path=/tmp/promtail_test_rTPtxgzxXg/logs/testFileRoll.log
2019/05/08 09:49:33 Seeked /tmp/promtail_test_rTPtxgzxXg/logs/testFileRoll.log - &{Offset:0 Whence:0}
level=info msg="stopped tailing file" path=/tmp/promtail_test_rTPtxgzxXg/logs/testFileRoll.log
level=info msg="start tailing file" path=/tmp/promtail_test_rTPtxgzxXg/logs/testFileRoll.log
2019/05/08 09:49:33 Seeked /tmp/promtail_test_rTPtxgzxXg/logs/testFileRoll.log - &{Offset:0 Whence:0}
2019/05/08 09:49:33 Seeked /tmp/promtail_test_rTPtxgzxXg/logs/testSymlinkRoll.log - &{Offset:0 Whence:0}
level=info msg="start tailing file" path=/tmp/promtail_test_rTPtxgzxXg/logs/testSymlinkRoll.log
2019/05/08 09:49:34 Re-opening moved/deleted file /tmp/promtail_test_rTPtxgzxXg/logs/testSymlinkRoll.log ...
2019/05/08 09:49:34 Successfully reopened /tmp/promtail_test_rTPtxgzxXg/logs/testSymlinkRoll.log
level=info msg="start tailing file" path=/tmp/promtail_test_rTPtxgzxXg/logs/testsubdir/testFile.log
2019/05/08 09:49:34 Seeked /tmp/promtail_test_rTPtxgzxXg/logs/testsubdir/testFile.log - &{Offset:0 Whence:0}
level=info file=/tmp/promtail_test_rTPtxgzxXg/logs/testFileRoll.log entry=roll0
level=info file=/tmp/promtail_test_rTPtxgzxXg/logs/testFileRoll.log entry=roll100
level=info file=/tmp/promtail_test_rTPtxgzxXg/logs/testFileRoll.log entry=roll101
level=info file=/tmp/promtail_test_rTPtxgzxXg/logs/testFileRoll.log entry=roll102
...
level=info file=/tmp/promtail_test_rTPtxgzxXg/logs/testFileRoll.log entry=roll197
level=info file=/tmp/promtail_test_rTPtxgzxXg/logs/testFileRoll.log entry=roll198
level=info file=/tmp/promtail_test_rTPtxgzxXg/logs/testFileRoll.log entry=roll199
--- FAIL: TestPromtail (26.25s)
    promtail_test.go:123: Timed out waiting for log entries:  still waiting for logs from /tmp/promtail_test_rTPtxgzxXg/logs/testFileRoll.log
```

--- a/pkg/promtail/promtail_test.go
+++ b/pkg/promtail/promtail_test.go
@@ -132,8 +132,8 @@ func TestPromtail(t *testing.T) {
 		t.Fatal("Could not delete a log file to verify metrics are removed: ", err)
 	}
 
-	// Sync period is 100ms in tests, need to wait for at least one sync period for tailer to be cleaned up
-	<-time.After(150 * time.Millisecond)
+	// Sync period is 500ms in tests, need to wait for at least one sync period for tailer to be cleaned up
+	<-time.After(500 * time.Millisecond)
 
 	//Pull out some prometheus metrics before shutting down
 	metricsBytes, contentType := getPromMetrics(t)
@@ -497,7 +497,9 @@ func buildTestConfig(t *testing.T, positionsFileName string, logDirName string) 
 	}
 	cfg.ScrapeConfig = append(cfg.ScrapeConfig, scrapeConfig)
 
-	cfg.TargetConfig.SyncPeriod = 10 * time.Millisecond
+	// Make sure the SyncPeriod is fast for test purposes, but not faster than the poll interval (250ms)
+	// to avoid a race between the sync() function and the tailers noticing when files are deleted
+	cfg.TargetConfig.SyncPeriod = 500 * time.Millisecond
 
 	return cfg
 }

--- a/pkg/promtail/targets/filetarget.go
+++ b/pkg/promtail/targets/filetarget.go
@@ -163,7 +163,7 @@ func (t *FileTarget) run() {
 		case event := <-t.watcher.Events:
 			switch event.Op {
 			case fsnotify.Create:
-				matched, err := filepath.Match(t.path, event.Name)
+				matched, err := doublestar.Match(t.path, event.Name)
 				if err != nil {
 					level.Error(t.logger).Log("msg", "failed to match file", "error", err, "filename", event.Name)
 					continue

